spec/unit/puppetlabs_spec_helper/tasks/module_inspect_spec.rb

```diff
@@ -0,0 +1,70 @@
+require 'spec_helper'
+
+describe 'rake module:inspect', type: :task, unless: puppet_3_or_older? do
+  let(:manifest_dir) { File.join(Dir.pwd, 'manifests') }
+  let(:no_results) { { 'host_class' => [], 'defined_type' => [] } }
+
+  before(:each) do
+    Dir.mkdir(manifest_dir)
+  end
+
+  context 'when there are no files' do
+    it 'runs without raising an error' do
+      expect { task.execute }.to output_json(no_results).to_stdout
+    end
+  end
+
+  context 'when there are empty manifests' do
+    before(:each) do
+      File.open(File.join(manifest_dir, 'empty.pp'), 'w') { |fd| }
+    end
+
+    it 'runs without raising an error' do
+      expect { task.execute }.to output_json(no_results).to_stdout
+    end
+  end
+
+  context 'when there are manifests containing class and defined types' do
+    before(:each) do
```

Comment:
Question: Instead of creating on the fly should we be using a fixtures directory instead?

Reply:
Would save the creating and deleting of files on-the-fly

Reply:
The task tests use an in memory filesystem, so the overhead of setting up and tearing down on the fly is negligible.